executor/linux_fork_server_executor.cpp

```diff
@@ -0,0 +1,585 @@
+#include "fuzzuf/utils/interprocess_shared_object.hpp"
+#include "fuzzuf/utils/errno_to_system_error.hpp"
+#include "fuzzuf/feedback/inplace_memory_feedback.hpp"
+#include "fuzzuf/feedback/exit_status_feedback.hpp"
+#include "fuzzuf/feedback/put_exit_reason_type.hpp"
+#include "fuzzuf/logger/logger.hpp"
+
+bool LinuxForkServerExecutor::has_setup_sighandlers = false;
+
+LinuxForkServerExecutor* LinuxForkServerExecutor::active_instance = nullptr;
+
+/**
+ * Precondition:
+ *   - A file can be created at path path_str_to_write_input.
+ * Postcondition:
+ *     - Run process below in order, with initializing members
+ *       * Configure signal handlers (Temporary workaround for non fork server mode. This will be removed in future.)
+ *       * Parsing and preprocessing of commandline arguments of PUT.
+ *       * Generate a file for sending input to  PUT.
+ *       * Configure shared memory
+ *         - afl_shm_size should indicate the size of shared memory which PUT built using AFL style cc uses.
+ *         - bb_shm_size should indicate the size of shared memory that is used to record Basic Block Coverage by PUT built using fuzzuf-cc.
+ *         - If both parameters are set to zero, it is considered as unused, and never allocate.
+ *         - In kernel, Both parameters are round up to multiple of PAGE_SIZE, then memory is allocated.
+ *       * Configure environment variables for PUT
+ *       * If fork server mode, launch fork server.
+ *       * NOTE: Executor does not take care about binding a CPU core. The owner fuzzing algorithm is responsible to it.
+ */
+LinuxForkServerExecutor::LinuxForkServerExecutor(  
+    const std::vector<std::string> &argv,
+    u32 exec_timelimit_ms,
+    u64 exec_memlimit,
+    const fs::path &path_to_write_input,
+    u32 afl_shm_size,
+    u32  bb_shm_size,
+    bool record_stdout_and_err,
+    std::vector< std::string > &&environment_variables_
+) :
+    Executor( argv, exec_timelimit_ms, exec_memlimit, path_to_write_input.string() ),
+    afl_edge_coverage(afl_shm_size),
+    fuzzuf_bb_coverage(bb_shm_size),
+
+    // cargv and stdin_mode are initialized at SetCArgvAndDecideInputMode
+    child_timed_out( false ),
+    record_stdout_and_err( record_stdout_and_err ),
+    // put_channel( "ipc:///tmp/forkserver.ipc" ) // FIXME: ユーザーが設定可能に
+    put_channel() // TODO: Make channel configurable outside of executor
+{
+    if (!has_setup_sighandlers) {
+	// For the time being, as signal handler is set globally, this function is static method, therefore it is not needed to set again if has_setup_handlers is ture.
+        SetupSignalHandlers();
+        has_setup_sighandlers = true;
+    }
+    // Assign self address to static pointer to make signal handlers can refer.
+    // active_instance may refer other instance if multiple LinuxForkServerExecutor exist, therefore following assertion is not appliable.
+    // assert(active_instance == nullptr);
+    active_instance = this;
```

Comment:
I think this part is no longer needed. This part is needed largely for alarm.

Reply:
Thank you for suggestion. I removed these lines.